### ISSUE TITLE
Bufferline issue

### ISSUE THREAD

**Issue by alechp** [abzcoding/lvim#79]
# Summary
- Having issue with Bufferline when I open a Cargo.toml file

#### Error Text
```
E5108: Error executing lua ...ack/packer/opt/bufferline.nvim/lua/bufferline/groups.lua:489: attempt to index local 'sublist' (a nil value)                        
stack traceback:                                                                                                                                                  
        ...ack/packer/opt/bufferline.nvim/lua/bufferline/groups.lua:489: in function 'sort_by_groups'                                                             
        ...ack/packer/opt/bufferline.nvim/lua/bufferline/groups.lua:509: in function 'render'                                                                     
        .../site/pack/packer/opt/bufferline.nvim/lua/bufferline.lua:108: in function <.../site/pack/packer/opt/bufferline.nvim/lua/bufferline.lua:99> 
```

#### Error screenshots
- When I open Cargo.toml after another file
> <img width="1904" alt="Screen Shot 2022-08-29 at 1 54 26 PM" src="https://user-images.githubusercontent.com/3058939/187297439-196e7de6-684e-41f3-9bf3-eb9bd6f4b929.png">

- What happens when I press enter
> <img width="1904" alt="Screen Shot 2022-08-29 at 1 54 49 PM" src="https://user-images.githubusercontent.com/3058939/187297467-dcdc422f-9f5b-478f-85f4-41fb558f4a95.png">

#### Problem 2: white highlight issue
- With this JS example, the active buffer looks normal
> <img width="1904" alt="Screen Shot 2022-08-29 at 2 01 13 PM" src="https://user-images.githubusercontent.com/3058939/187298120-a98442e6-5482-415a-a1b2-8b9101abbf92.png">

- With Rust example, active buffer has a strange white highlight
> <img width="1904" alt="Screen Shot 2022-08-29 at 1 56 05 PM" src="https://user-images.githubusercontent.com/3058939/187297653-a844cd16-ccf4-4cbf-9f4a-d21cbad73f0c.png">

**Comment by luxus** [comment]
<img width="949" alt="Screenshot 2022-08-30 at 00 17 47" src="https://user-images.githubusercontent.com/7449/187308754-2476bc81-2876-42ab-97da-6f7da0c50efd.png">
i can confirm a issue when open a cargo.toml

**Comment by abzcoding** [comment]
can you guys do these?

1. first make sure bufferline is up to date

```shell
cd ~/.local/share/lunarvim/site/pack/packer/opt/bufferline.nvim
git checkout main
git pull
```

2. then run 
```vim
:TSUpdate
```
 inside lvim

**Comment by abzcoding** [comment]
> yes it fixed it.. bufferline was stuck on a specific commit

yeah I'm not sure if it's a packer issue or a lunarvim issue, but I'm using this command to keep the plugins up to date
```shell
# run the following only once to fix the issue
find ~/.local/share/lunarvim/site/pack/packer -maxdepth 2 -mindepth 2 -type d -exec git -C {} checkout main \;
find ~/.local/share/lunarvim/site/pack/packer -maxdepth 2 -mindepth 2 -type d -exec git -C {} checkout master \;

# this will keep lunarvim up to date
git -C ~/.local/share/lunarvim/lvim pull

# use this command to keep them up to date
find ~/.local/share/lunarvim/site/pack/packer -maxdepth 2 -mindepth 2 -type d -exec git -C {} pull \;
```

**Comment by crisidev** [comment]
So you don't really use `LvimUpdate` right? Because that is going to revert stuff to a specific commit..

**Comment by abzcoding** [comment]
> So you don't really use `LvimUpdate` right? Because that is going to revert stuff to a specific commit..

yep I don't use it atm, I should probably create an issue for it, or even fix it, but sadly I'm busy rn

**Comment by luxus** [comment]
can someone patch the `~/.local/bin/lvim` script and add a `lvim --update`?
i'm a litte busy too :D

**Comment by abzcoding** [comment]
@alechp, the highlight issue can be fixed as well
check 
https://github.com/abzcoding/lvim/blob/a98f8e37b0f228e43108dd3771a3187a4f1ba30a/lua/user/bufferline.lua#L6-L8

and also
https://github.com/abzcoding/lvim/blob/a98f8e37b0f228e43108dd3771a3187a4f1ba30a/lua/user/bufferline.lua#L162-L163

**Comment by abzcoding** [comment]
> Adding the other update commands to aliases. Thanks Abz!
> ```

you're welcome <3

**Comment by abzcoding** [comment]
Also, everybody, please be advised that **using that command means breaking stuff**

Since the normal `lvim` keeps a working commit of those plugins in this file https://github.com/LunarVim/LunarVim/blob/rolling/snapshots/default.json

So, please use with caution

**Comment by alechp** [comment]
> @alechp, the highlight issue can be fixed as well check
> 
> https://github.com/abzcoding/lvim/blob/a98f8e37b0f228e43108dd3771a3187a4f1ba30a/lua/user/bufferline.lua#L6-L8

That did the trick! 
<img width="893" alt="Screen Shot 2022-08-30 at 4 50 06 PM" src="https://user-images.githubusercontent.com/3058939/187563361-68f8f43b-c780-4a6f-a4f8-5bbd8fae3d06.png">

**Comment by alechp** [comment]
> @alechp, the highlight issue can be fixed as well check
> 
> https://github.com/abzcoding/lvim/blob/a98f8e37b0f228e43108dd3771a3187a4f1ba30a/lua/user/bufferline.lua#L6-L8
> 
> and also
> 
> https://github.com/abzcoding/lvim/blob/a98f8e37b0f228e43108dd3771a3187a4f1ba30a/lua/user/bufferline.lua#L162-L163

Thanks Abz

Ended up being resolved by adding this to tmux config:
```
set -g default-terminal "tmux-256color"
set -ag terminal-overrides ",xterm-256color:RGB"
``` 

<img width="904" alt="Screen Shot 2022-08-30 at 5 20 20 PM" src="https://user-images.githubusercontent.com/3058939/187566078-11f98ad6-e147-4178-814c-3f02cf613e09.png">

---

Extra context:

- Dug deeper and found out that this was only happening in zsh (default term for tmux), not in nushell (default terminal)
- My pipeline is kitty->tmux->zsh->lvim
- Was a color issue, fixed by this:
> https://gist.github.com/andersevenrud/015e61af2fd264371032763d4ed965b6

**Comment by alechp** [comment]
## Update
The bufferline bug seems to return randomly (assuming head gets updated after each packer sync or re-compile, but didn't verify that's the trigger)

Quick fix:
```
cd ~/.local/share/lunarvim/site/pack/packer/opt/bufferline.nvim
git checkout main && git pull
```

<img width="874" alt="Screen Shot 2022-09-07 at 10 03 27 AM" src="https://user-images.githubusercontent.com/3058939/188938290-af4c160f-b9e9-4a1a-9c82-2154070aab49.png">

**Comment by abzcoding** [comment]
> ## Update
> The bufferline bug seems to return randomly (assuming head gets updated after each packer sync or re-compile, but didn't verify that's the trigger)

yeah it seems like running a packer sync or install forces back the versions inside [default.json](https://github.com/LunarVim/LunarVim/blob/rolling/snapshots/default.json)

lunarvim itself has been [updated](https://github.com/LunarVim/LunarVim/commit/e485cd6e406b8ba8e5d02a87db954a1d924036ce) to use the latest bufferline ( 0b4b86) 

so hopefully this will not happen anymore

**Comment by alechp** [comment]
Looks good! Updated lunarvim on rolling and re-synced. Checked commit hashes and it seems like HEAD matching main afterall (checking out main updated head to `0b4b86` is what fixed it in previous screenshot; confirmed that both detached head and main branch set to 0b4b86 🥳 )

![Screen Shot 2022-09-07 at 12 20 36 PM](https://user-images.githubusercontent.com/3058939/188960150-66d72fa0-6b35-4213-ae93-ec4c10c31bdb.png)

MIght be too soon to celebrate, but looks good so far. Thanks for all your help @abzcoding 🤝
